Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners_test.py

```diff
@@ -199,55 +176,135 @@ def test_aggregate(owners, expected_out):
-            {'Name': 'aa', 'Email': '', 'Source': 'source3 | source4', 'Timestamp': '4', 'Count': 2},
-        ],
-        [
-            {'Name': 'aa', 'Email': 'email1@gmail.com', 'Source': 'source1 | source2', 'Timestamp': '2', 'Ranking Score': 1.0},
-            {'Name': 'aa', 'Email': '', 'Source': 'source3 | source4', 'Timestamp': '4', 'Ranking Score': 1.0},
-        ]
-    ),
-    # unequal counts
-    (
-        [
-            {'Name': 'aa', 'Email': 'email1@gmail.com', 'Source': 'source1', 'Timestamp': '2', 'Count': 1},
-            {'Name': 'aa', 'Email': '', 'Source': 'source3 | source4', 'Timestamp': '4', 'Count': 2},
-        ],
-        [
-            {'Name': 'aa', 'Email': 'email1@gmail.com', 'Source': 'source1', 'Timestamp': '2', 'Ranking Score': 0.5},
-            {'Name': 'aa', 'Email': '', 'Source': 'source3 | source4', 'Timestamp': '4', 'Ranking Score': 1.0},
-        ]
-    ),
-    # empty owners
-    (
-        [],
-        []
-    )
+@pytest.mark.parametrize('scores,expected_out', [
+    ([0.5, 1], [0.67, 0.83]),
+    ([1, 10, 100], [.5, .55, .95]),
+    ([], []),
 ])
-def test_score(deduplicated, expected_out):
-    assert score(deduplicated) == expected_out
+def test_normalize_scores(scores, expected_out):
+    assert np.allclose(normalize_scores(scores), expected_out, atol=0.01)
+
+
+def test_score_base_case(mocker):
+    """
+    Test the default scoring case
+    """
+    model_mock = Mock()
+    model_mock.predict.return_value = np.array([4, 7, 31, 0.33, 15])
+
+    mocker.patch(
+        'RankServiceOwners.load_pickled_xpanse_object',
+        return_value=model_mock
+    )
+
+    asm_system_ids = ["afr-rdp-1", "j291mv-is"]
+
+    owners = [
+        {'Name': 'Amira', 'Email': 'amira@example.com', 'Source': 'GCP | \
+            Chain: GCP project owner of a service account attached to the VM | \
+                Owner-In-Tags-From-PrismaCloud | Owner-In-Tags-From-GCP', 'Timestamp': '1'},
+        {'Name': 'Brandon', 'Email': 'brandon@example.com', 'Source': 'GCP | \
+            Owner-In-Tags-From-GCP | Tenable.io | New-Log-Source', 'Timestamp': '2'},
+        {'Name': 'Chun', 'Email': 'chun@example.com', 'Source': 'SNOW-CMDB', 'Timestamp': '3'},
+        {'Name': 'Divya', 'Email': 'divya@example.com', 'Source': 'Chain: Chain: \VM launches with a service account, \
+            which belongs to GCP project my-project that grants Editor permissions to svc-acct@my-project.gserviceaccount.com, \
+                which this person can impersonate', 'Timestamp': '4'},
+        {'Name': 'Automation First Remediation', 'Email': 'afr@example.com', 'Source': 'GCP | Splunk', 'Timestamp': '5'},
+    ]
+    scores_out = [owner['Ranking Score'] for owner in score(asm_system_ids=asm_system_ids, owners=owners)]
+    assert np.allclose(scores_out, [0.53, 0.56, 0.77, 0.5, 0.63])
+
+
+def test_score_model_load_fail(mocker):
```

Comment:
can this be simplified? the `owners` seem very complex if you just want to verify that if `load_pickled_xpanse_object` raises an exception, we still get an output.